cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -184,158 +173,220 @@ public void launchTask(final ExecutorDriver driver, final TaskInfo task) {
             final SlaveStatusDetails details = SlaveStatusDetails.newBuilder()
                 .setStatusDetailsType(SlaveStatusDetails.StatusDetailsType.ERROR_DETAILS)
                 .setSlaveErrorDetails(
-                    SlaveErrorDetails.newBuilder()
-                        .setMsg(msg)
-                        .setDetails(e.getMessage())
+                        SlaveErrorDetails.newBuilder()
+                                .setMsg(msg)
+                                .setDetails(e.getMessage() != null ? e.getMessage() : "-")
                 )
-                .build();
+                    .build();
             final TaskStatus taskStatus = taskStatus(task, TaskState.TASK_ERROR, details);
             driver.sendStatusUpdate(taskStatus);
         }
         LOGGER.debug(taskIdMarker, "< launchTask(driver : {}, task : {})", driver, protoToString(task));
     }
 
+    private void startJob(ExecutorDriver driver, TaskInfo task, TaskDetails taskDetails) {
+        if (process == null) {
+            driver.sendStatusUpdate(taskStatus(task, TaskState.TASK_ERROR, nullSlaveStatusDetails()));
+            return;
+        }
+
+        NodeJobTask nodeJob = taskDetails.getNodeJobTask();
+        AbstractNodeJob job;
+        switch (nodeJob.getJobType()) {
+            case REPAIR:
+                job = new NodeRepairJob(task.getTaskId());
+                break;
+            case CLEANUP:
+                job = new NodeCleanupJob(task.getTaskId(), executorService);
+                break;
+            default:
+                return;
+        }
+        if (startJob(job)) {
+            driver.sendStatusUpdate(taskStatus(task, TaskState.TASK_RUNNING, nullSlaveStatusDetails()));
+        } else {
+            driver.sendStatusUpdate(taskStatus(task, TaskState.TASK_ERROR, nullSlaveStatusDetails()));
+        }
+    }
+
+    private void jobStatus(ExecutorDriver driver, TaskInfo task) {
+        if (process == null) {
+            LOGGER.error("No Cassandra process to handle node-job-status");
+            if (task != null)
+                driver.sendStatusUpdate(taskStatus(task, TaskState.TASK_ERROR, nullSlaveStatusDetails()));
```

Comment:
Also this entry assertion is almost exactly the same as what's done in `startJob`. Are (can) they actually be the same and be factored into a helper method to avoid duplication.